tests/generative/variable_strategies.py

```diff
@@ -58,8 +60,37 @@ def variable(patient_tables, event_tables, schema, value_strategies):
     #     permitted to return a one-row-per-patient series, because such series can always be
     #     composed a many-rows-per-patient series; so there are series strategy functions that,
     #     always or sometimes, ignore the frame argument.
+
+    # Max depth
+    #
+    # The depth_exceeded functions force hypothesis to return a terminating node if the current
+    # test depth is too deep.
+    # Otherwise, the generated graph can continue forever, and will eventually hit the
+    # hypothesis limit (100) and will be abandoned. This results in too many invalid examples,
+    # which triggers the too-many-filters healthcheck.
+    #
+    # If the max limit is set high - e.g. if we always let it go to 100 and then return our
+    # default terminating node, generating the examples takes a really long time.
+    #
+    # Setting the depth at which a table strategy bails too low means that hypothesis takes
+    # too long to shrink examples.
+    #
+    # The defaults are therefore set, somewhat arbitrarily, to 30 for series and 75 for tables.
+
+    def depth_exceeded(max_depth=None):
+        max_depth = max_depth or environ.get("GENTEST_SERIES_MAX_DEPTH", 30)
```

Comment:
I think you need an `int` cast around the `environ.get()` here. But might be easier to do as module level constants? e.g. at the top of the module:
```py
SERIES_MAX_DEPTH = int(environ.get("GENTEST_SERIES_MAX_DEPTH", 30))
TABLE_MAX_DEPTH = int(environ.get("GENTEST_TABLE_MAX_DEPTH", 75))

Reply:
I've done this, but I've also changed my mind about there needing to be 2 different settings, and after a bit more systematic checking, I don't think it makes a difference.  In fact, since the check is only used for a filtered/sorted table, and both of those strategies always use a series at least once, I think they'll always use the series depth anyway